cmake/toolchain/espressif/generic.cmake

```diff
@@ -12,20 +12,29 @@ set(BINTOOLS gnu)
 
 file(GLOB toolchain_paths
   LIST_DIRECTORIES true
-  ${TOOLCHAIN_HOME}
+  ${TOOLCHAIN_HOME}/*-esp32*/*-elf
+  ${TOOLCHAIN_HOME}/*-esp-elf
+
   )
 
 if(toolchain_paths)
-  list(GET toolchain_paths 0 soc_toolchain_path)
+  list(GET toolchain_paths 0 some_toolchain_path)
+
+  get_filename_component(one_toolchain_root "${some_toolchain_path}" DIRECTORY)
+  get_filename_component(one_toolchain "${some_toolchain_path}" NAME)
 
-  get_filename_component(soc_toolchain "${soc_toolchain_path}" NAME)
+  set(CROSS_COMPILE_TARGET ${one_toolchain})
+  set(SYSROOT_TARGET	   ${one_toolchain})
+endif()
 
-  set(CROSS_COMPILE_TARGET ${soc_toolchain})
-  set(SYSROOT_TARGET       ${soc_toolchain})
+if(NOT CROSS_COMPILE_TARGET)
+  message(FATAL_ERROR
+    "Unable to find toolchain in ${TOOLCHAIN_HOME} "
+    "Run `west espressif install` to download it. Then, export to path accordingly.")
```

Comment:
@tejlmand This is current error if toolchain not found by either missing its installation or having previous installation version.

Reply:
yes, I see that.
But this just ask user to re-install, but a user already having the toolchain installed should not need to re-install it.